allennlp/common/registrable.py

```diff
@@ -38,18 +38,22 @@ class Registrable(FromParams):
     a subclass to load all other subclasses and the abstract class).
     """
 
-    _registry: Dict[Type, Dict[str, Type]] = defaultdict(dict)
+    _registry: Dict[Type, Dict[str, Tuple[Type, str]]] = defaultdict(dict)
     default_implementation: str = None
 
     @classmethod
-    def register(cls: Type[T], name: str, exist_ok=False):
+    def register(cls: Type[T], name: str, constructor: str = None, exist_ok: bool = False):
         """
         Register a class under a particular name.
```

Comment:
Now we can have markdown in docstrings, can you add a couple of examples of registering different constructors and their corresponding jsonnet config?